doc/tutorials/specification-language.md

```diff
@@ -0,0 +1,97 @@
+# Specification Language
+
+This tutorial introduces you to the concepts of the specification language which are used by Elektra. The specification language
+can be used to apply a validation for the configuration that you want to ship for your application.
+
+## How Elektra perfroms validation
+
+One of the main concepts of Elektra are [namespaces](namespaces.md) which is also used by the specification language.
+When you [mount](mount.md) a configuration such as `kdb mount test.conf /tests dump` you usually use one of the four standard namespaces
+`user`, `dir`, `proc` or `system`.
+
+There is a 5th namespace called the `spec` namespace and has a special behavior. Any configuration you mount under `spec` will
+be used to validate configurations in other namespaces under the same path. So in the following example you can write any validation
+data in the `test.spec` file and can be assured it will be validated in the other namespaces:
+
+```shell script
+# Mounts the specification 'test.spec'
+# Configuration validation is now applies to all keys below the '/tests' path
+kdb mount test.spec spec/tests dump
+
+# An example configuration which will by default gets mounted to the 'user' namespace
+kdb mount test.conf /tests dump type
+```
+
+Elektra takes all configuration settings which the user or admin provides and parses it into a structured internal format.
+This structure is called a `KeySet` and is format agnostic (meaning that validation can be applied to any file format
+such as `ini`, `yml`, etc.). After successful parsing into the KeySet, Elektra then passes this KeySet to various
+validation plugins. You can get a picture of all plugins [here](/src/plugins). Every plugin performs validation
+of the KeySet depending on the metadata provided (more detail about this will come in the next section) and emits warnings
+or errors. If any error happened, the changes of the configuration settings are not applied.
+
+E.g. you can see
+that in the upper example we also mounted the [type](/src/plugins/type/README.md) plugin for validation checks. This for example
+can check for wrong types:
+
+```shell script
+kdb meta-set spec/tests/bool 'type' boolean
+kdb set /tests/bool Batman
```

Comment:
Does the shell recorder also work for non-connected code snippets?

Reply:
@sanssecours can you help out here? (Maybe update the docu of shell recorder)

Reply:
> Does the shell recorder also work for non-connected code snippets?

In theory it might work. I never tried it myself though.

> @sanssecours can you help out here? (Maybe update the docu of shell recorder)

Sorry, I do not think that I have time to do that in the moment.

Reply:
If it works, there would an extra "cleanup" section be needed where you unmount everything and cleanup all created tickets. This though is more confusing for a new reader and just pumps up the tutorial in terms of lines. I would prefer to spare that and execute everything local before merging it into master to avoid typos

Reply:
This cleanup also needs to be done on the computer where a reader typed in the commands, so I do not see any problems with having this.

Local verification if a tutorial works, however, has many problems and it was often shown that it does not work because bugs in tutorials usually only lead to frustrations and not to bug reports, e.g. see #3207

Reply:
@Piankero You can take a look at doc/tutorials/install-config-files.md

> Does the shell recorder also work for non-connected code snippets?

Yes, I did this in install-config-files.

>  there would an extra "cleanup" section be needed where you unmount everything and cleanup all created keys

Correct and it is a bit annoying to get this one correct. You have to be very careful that you don't start your test runs in an environment that is messed up from previous tests.

When I spent so much time on the cleanup only I felt stupid, but just yesterday my own shell recorder tests saved me from committing a massive behavior change for the merge without noticing.